src/main/java/uk/gov/companieshouse/company/links/service/CompanyInsolvencyService.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.company.links.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.http.ApiKeyHttpClient;
+import uk.gov.companieshouse.api.http.HttpClient;
+import uk.gov.companieshouse.api.insolvency.CompanyInsolvency;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.company.links.exception.RetryableErrorException;
+import uk.gov.companieshouse.logging.Logger;
+
+@Service
+public class CompanyInsolvencyService extends BaseApiClientService {
+
+    @Value("${api.company-insolvency-api-key}")
+    private String companyInsolvencyApiKey;
+
+    @Value("${api.endpoint}")
+    private String companyInsolvencyApiUrl;
+
+    /**
+     * Construct a company insolvency service - used to retrieve a company insolvency record.
+     * @param logger the CH logger
+     */
+    @Autowired
+    public CompanyInsolvencyService(Logger logger) {
+        super(logger);
+    }
+
+    /**
+     * Retrieve a company insolvency given a company number from company-profile-api.
+     *
+     * @param companyNumber the company's company number
+     * @return an ApiResponse containing the CompanyInsolvencyApi data model
+     */
+    public ApiResponse<CompanyInsolvency> getCompanyInsolvency(String contextId,
+                                                               String companyNumber)
+            throws RetryableErrorException {
+        logger.trace(String.format("Call to GET company insolvency with contextId %s "
+                        + "and company number %s", contextId, companyNumber));
+
+        String uri = String.format("/company/%s/insolvency", companyNumber);
+        return executeOp(contextId, "getCompanyInsolvencyApi", uri,
+            getApiClient(contextId)
+                    .privateDeltaInsolvencyResourceHandler().getInsolvency(uri));
+    }
+
+    /**
+     * Get an internal api client instance.
+     */
+    public InternalApiClient getApiClient(String contextId) {
```

Comment:
Better use Supplier bean like we did in charges-delta-consumer and company-metrics-consumer instead of these two methods?

Reply:
Relevant code change done